### PR TITLE
Clarify repositories structure in install kubeadm document

### PR DESCRIPTION
**Pull request by xmudrii**
We received feedback that it's not clear enough that there's a dedicated package repository for each Kubernetes minor version (see #42810). This is a regression from #43407 and this PR is supposed to address this concern.

/assign @sftim @saschagrunert @cpanato @jeremyrickard 
cc @kubernetes/release-engineering

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (Affan-7) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:195]: I wonder if this statement will actually help a new user to get the repository that they want.

**Comment by xmudrii** [comment]
I found a formatting issue in instructions for Red Had-based distros:

image

The issue is also present on k8s.io: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/

I fixed this issue in the latest commit.

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (Affan-7) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:170]: Can you share some more specific details from where the user can find the installation guide for their desired minor version.

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (Affan-7) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:243]: Also this one.

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md]: Fixed in the latest commit

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md]: Fixed in the latest commit

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md]: Fixed in the latest commit

**Comment by xmudrii** [comment]
@Affan-7 I addressed your comments in the latest commit
/hold cancel

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:177]:
```suggestion
your desired minor version. The official Kubernetes package repositories provide downloads for Kubernetes versions starting
```

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:203]:
```suggestion
   change the Kubernetes minor version in the URL to match your desired minor version
   (you should also check that you are reading the documentation for the version of Kubernetes
   that you plan to install).
```

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:251]:
```suggestion
   in URLs to match your desired minor version
   (you should also check that you are reading the documentation for the version of Kubernetes
   that you plan to install).
```

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (Affan-7) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:24]: Instead of using this list we can add a similar note similar to this one.

https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-binary-with-curl-on-linux

This can be added before the installation steps.

Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:24]: I copied this code from [the Upgrading kubeadm clusters page](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/) but I see that it's also affected by the issue that you mentioned. Let's leave it as it is for now and I can create a follow-up issue once this PR is merged.

**Comment by xmudrii** [comment]
@sftim Addressed your comments, PTAL again

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md
Comment (sftim) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:24]: > Instead of using this list we can add a note similar to this one.
> 
> https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-binary-with-curl-on-linux

We could, but let's not.

We want to help readers find the right docs for their version (we _don't_ want to set an expectation that the Kubernetes website covers older versions). Let's avoid trapping ourselves into an expectationof backwards compatibility.

**Review by sftim — COMMENTED** [review]
LGTM with the caveat from https://github.com/kubernetes/website/pull/43458/files#r1357008050

/lgtm

**Comment by sftim** [comment]
/lgtm

Ideally: squash to fewer commits

**Comment by xmudrii** [comment]
@sftim Squashed the PR to one commit, PTAL again

**Comment by sftim** [comment]
Yep, still looks good to me (Prow has kept the **lgtm** label).

**Review by aj11anuj — COMMENTED** [review]
For that skew version list thing, a separate issue can be raised later on. Rest everything
LGTM, 👍

**Comment by tengqm** [comment]
Thanks.
/approve

**Pull request merged**
